### ISSUE TITLE
Obsolete tracks in music library after "unshare from self"

### ISSUE THREAD

**Issue by paulijar** [owncloud/music#567]
If user A shares an audio file with user B and then unshares it, the track appears and disappears from the B's music library as expected. If, instead, B unshares the audio file shared by A, then the track is left in B's library. The unshared track cannot be played by B though, and if it was used to provide embedded album art, then the album cover image is missing. There is no way for B to get rid of the obsolete track reference, expect by rebuilding the whole music library.

The behavior is as described above with the latest development version of the Music app. With older versions, the obsolete track references used to prevent the music library from loading at all.

**Comment by paulijar** [comment]
The commit c03de5b together with PR  owncloud/core#28337 solves this issue. I'm still considering if there would be some clever way to cleanup the obsolete track references (say, in background task) when working on top of older core which does not have the required fix.

**Comment by paulijar** [comment]
For Nextcloud, the relevant PR is https://github.com/nextcloud/server/pull/5797.

**Comment by paulijar** [comment]
I added an option `--clean-obsolete` to the `occ` command `music:scan` which can be used to get rid of any obsolete, previously unshared tracks in the music library. This is potentially a heavy-weight operation, and hence, it cannot be run automatically in the background cleanup task.

As the core-side fixes for this issue have been merged both into ownCloud and Nextcloud, and the inconsistent library can now be fixed by the `occ` tool, I think that this issue has been addressed as well as possible. Closing.
